mlflow/resource_experiment.go

```diff
@@ -0,0 +1,160 @@
+package mlflow
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/databrickslabs/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// MLFLowExperiment defines the parameters that can be set in the resource.
+type MLFLowExperiment struct {
+	Name        string `json:"name"`
+	Tags        []Tag  `json:"tags,omitempty"`
+	Description string `json:"description,omitempty"`
+}
+
+// MLFLowExperimentAPI defines the response object from the API
+type MLFLowExperimentAPI struct {
+	ExperimentId     string `json:"experiment_id"`
+	Name             string `json:"name"`
+	ArtifactLocation string `json:"artifact_location,omitempty"`
+	LifecycleStage   string `json:"lifecycle_stage,omitempty"`
+	LastUpdateTime   int64  `json:"last_update_time,omitempty"`
+	CreationTime     int64  `json:"creation_time,omitempty"`
+	Tags             []Tag  `json:"tags,omitempty"`
+}
+
+type MLFLowExperimentsAPI struct {
+	Experiment MLFLowExperimentAPI `json:"experiment"`
+}
+
+func (d *MLFLowExperiment) toAPIObject(schema map[string]*schema.Schema, data *schema.ResourceData) (*MLFLowExperimentAPI, error) {
+	// Extract from ResourceData.
+	if err := common.DataToStructPointer(data, schema, d); err != nil {
```

Comment:
DataToStructPointer already interacts with ResourceData, no need to set it directly.

CRUD methods are short enough to be inlined to ExperimentResource, so please rewrite and follow conventions of other resources in the codebase.